mutiny-core/src/lsp/lsps.rs

```diff
@@ -0,0 +1,228 @@
+            pending_buy_requests: Arc::new(Mutex::new(HashMap::new())),
+            pending_channel_info: Arc::new(Mutex::new(HashMap::new())),
+        };
+
+        let events_client = client.clone();
+        utils::spawn(async move { events_client.handle_events().await; });
+
+        Ok(client)
+    }
+
+    pub(crate) async fn handle_events(&self) {
+        loop {
+            for event in self.liquidity_manager.get_and_clear_pending_events() {
+                match event {
+                    events::Event::LSPS2(JITChannelEvent::GetInfoResponse { channel_id, opening_fee_params_menu, user_channel_id, min_payment_size_msat, max_payment_size_msat, ..  }) => {
+                        
+                        log_debug!(
+                            self.logger,
+                            "received get info response!"
+                        );
+
+                        let mut pending_fee_requests = self.pending_fee_requests.lock().unwrap();
+
+                        if let Some(fee_response_sender) = pending_fee_requests.remove(&user_channel_id) {
+                            if let Err(_e) = fee_response_sender.send(GetInfoResponse {
+                                channel_id,
+                                opening_fee_params_menu,
+                                min_payment_size_msat,
+                                max_payment_size_msat,
+                                user_channel_id,
+                            }) {
+                                // log error
+                            }
+                        }
+                    },
+                    events::Event::LSPS2(JITChannelEvent::InvoiceGenerationReady { scid, cltv_expiry_delta, user_channel_id, counterparty_node_id, payment_size_msat, .. }) => {
+                        let mut pending_buy_requests = self.pending_buy_requests.lock().unwrap();
+
+                        if let Some(buy_response_sender) = pending_buy_requests.remove(&user_channel_id) {
+
+                            let invoice_expiry_delta_secs = 60 * 60 * 24 * 3;
+                            let (payment_hash, payment_secret) = self
+                                .channel_manager
+                                .create_inbound_payment(None, invoice_expiry_delta_secs, None)
+                                .unwrap();
+
+                            let lsp_route_hint = RouteHint(vec![RouteHintHop {
+                                src_node_id: counterparty_node_id,
+                                short_channel_id: scid,
+                                fees: RoutingFees { base_msat: 0, proportional_millionths: 0 },
+                                cltv_expiry_delta: cltv_expiry_delta.try_into().unwrap(),
+                                htlc_minimum_msat: None,
+                                htlc_maximum_msat: None,
+                            }]);
+
+                            let mut invoice = InvoiceBuilder::new(self.network.into())
```

Comment:
would it not be better to use `create_invoice_from_channelmanager_and_duration_since_epoch` here?

Reply:
can't because we need to add the lsp route hint